tests/ui/feature-gates/feature-gate-explicit-extern-abis.rs

```diff
@@ -0,0 +1,7 @@
+//@ run-pass
+//@ run-rustfix
+
+extern fn _foo() {}
+//~^ WARN extern declarations without an explicit ABI are deprecated
+
+fn main() {}
```

Comment:
This test should be `check-pass` rather than `run-pass`.

Please add a comment at the top describing the purpose of the test and the expected behavior, as outlined below.  See `feature-gate-public_private_dependencies.rs` for an example of that for another test that doesn't follow the usual pattern.

Please use `//@ revisions` to test that:

- Both with and without the feature flag, it only warns in current editions (and that rustfix works as intended).
- That in the future edition and without the feature flag, it still only warns.
- That in the future edition with the feature flag, it gives the error.

That way this test will more usefully exercise the edges around the feature flag handling.

@rustbot author